src/TsTypes.ts

```diff
@@ -58,54 +61,99 @@ export namespace TsType {
   export interface TsProp {
     name: string
     required: boolean
-    type: TsType
+    type: TsTypeBase
   }
 
-  export class Any extends TsType {
+  export class Any extends TsTypeBase {
     _type() {
       return 'any'
     }
   }
-  export class String extends TsType {
+  export class String extends TsTypeBase {
     _type() {
       return 'string'
     }
   }
-  export class Boolean extends TsType {
+  export class Boolean extends TsTypeBase {
     _type() {
       return 'boolean'
     }
   }
-  export class Number extends TsType {
+  export class Number extends TsTypeBase {
     _type() {
       return 'number'
     }
   }
-  export class Object extends TsType {
+  export class Object extends TsTypeBase {
     _type() {
       return 'Object'
     }
   }
-  export class Void extends TsType {
+  export class Void extends TsTypeBase {
     _type() {
       return 'void'
     }
   }
-  export class Literal extends TsType {
+  export class Literal extends TsTypeBase {
     constructor(private value: any) { super() }
     _type() {
       return this.value
     }
   }
 
-  export class Array extends TsType {
-    constructor(private type?: TsType) { super() }
+  export class EnumValue {
+    identifier: string
+    value: string
+
+    constructor(enumValues: string[]) {
+      let hasValue = !!enumValues[0]
```

Comment:
We don't generate string identifiers.  We generate literal identifiers.  So this would be a very special case for a not very often used item, and I'm not sure how we would differentiate it...  Our output is:

```
export enum Bar {
  One = 1,
  Two = 2,
  Three = 3
}
```

Not

```
export enum Bar {
  "One" = 1,
  "Two" = 2,
  "Three" = 3
}
```

Really we need to be worried about what the schema supports--and we support strings or integers as enum member values.  If they added `''` as a value in a string enum, it would work fine.  But as far as our own proprietary extension for the names goes, we can declare whatever requirements we want on those values.  An empty string is a special case here (as we don't support any other quoted literal); do you think it's worth the effort?

Although I suppose if you had `"''"` (a double-quoted set of single-quotes) as the `tsEnumName` in the schema, that might print an epty string as the identifier...